experimenter/experimenter/jetstream/tests/test_tasks.py

```diff
@@ -598,12 +594,16 @@ def read(self):
         def open_file(filename):
             return File(filename)
 
-        mock_open.side_effect = open_file
-        mock_exists.return_value = True
+        with (
+            patch("experimenter.jetstream.client.default_storage.open") as mock_open,
+            patch("experimenter.jetstream.client.default_storage.exists") as mock_exists,
+        ):
+            mock_open.side_effect = open_file
+            mock_exists.return_value = True
```

Comment:
You can provide patch() with the function to replace it with / the return value

```suggestion
         with (
            patch("experimenter.jetstream.client.default_storage.open", open_file),
            patch("experimenter.jetstream.client.default_storage.exists", side_effect=True)
        ):
```

Reply:
Oh yeah good suggestions.  I'll just leave it as it is for now, but this is good to know in future.